### ISSUE TITLE
Your Partial Classes Documentation doesn't show how to use [Parameter] tag

### ISSUE THREAD

**Issue by DataJuggler** [aspnet/AspNetCore.Docs#16399]
I love that I can move code to partial class now, but I can't find any information about what namespace to add to make a property with a [Parameter] tag be available in the partial class.

It isn't obvious what namespace this is.

Thanks, 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: be4c4d0c-149c-bde1-c7fa-f082916390ca
* Version Independent ID: 9d945996-ad02-bd4b-58ea-c465cc5cc74c
* Content: [Create and use ASP.NET Core Razor components](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/components.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/components.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @DataJuggler ... The namespace rules are in [Import components](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1#import-components) section with introductory remarks in the [Component classes](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1#component-classes) section (last paragraph). The point on using custom folders is also mentioned in the *Import components* section in a sentence under the bullet list).

Most authored components will follow the 3rd bullet point in the *Import components* section, which is why the example shown in the [Partial class support](https://docs.microsoft.com/en-us/aspnet/core/blazor/components?view=aspnetcore-3.1#partial-class-support) section has the namespace `BlazorApp.Pages`. It goes by the **_folder_** where the component exists.

If the `currentCount` field is a parameter in the example, the parameter's fully qualified reference is the namespace, class name, and parameter name: `BlazorApp.Pages.Counter.CurrentCount`. That would the same construction regardless of whether the parameter is located in the partial class (`Counter`) or in an `@code` block in the *Counter.razor* file ... they have the same namespace (`BlazorApp.Pages`) and class (`Counter`).

What I'd like to do for this (thus far) is state explicitly in the *Component classes* section (last paragraph) the typical scenario, which is essentially what the 3rd bullet point states in the *Import components* section. The *Component classes* paragraph launches right into using a custom folder without first mentioning that the component namespace is derived from its location (folder).

Did you have a special scenario not covered by the current content, or was there something specifically confusing about the text that I've mentioned?

Even if you're ok now, leave this issue open ... it will close automatically when the PR merges.

**Comment by DataJuggler** [comment]
Someone from Stackoverflow posted a link where I figured it out.

I had all these namespaces:
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Components.Authorization;
using Microsoft.AspNetCore.Components.Forms;
using Microsoft.AspNetCore.Components.Routing;
using Microsoft.AspNetCore.Components.Web;

But I didn't have the root of AspNetCore.Components:
using Microsoft.AspNetCore.Components;

Thanks for the response, and sorry I posted. All the examples I found used @Code inline on the .razor page.

**Comment by guardrex** [comment]
That shouldn't be necessary with 3.1 to use partial classes. That was a requirement for 3.0. Are you on 3.1?

**Comment by DataJuggler** [comment]
Yes I am using 3.1. If II create a partial class and do not add a using statement for Microsoft.AspNetCore.Components, I cannot have a [Parameter] tag on the page.

I have the latest Visual Studio installed 16.4.2.

Everything works fine as long as I add the using reference line.

**Comment by guardrex** [comment]
Oh! ... I see what you mean. Yes, `ParameterAttribute` is in the `Microsoft.AspNetCore.Components` namespace. Yes, I wonder if it would be best if the templates had that imported.

Let's ping @danroth27 here to see if an engineering issue should be opened for it:

**TL;DR** The question is should `using Microsoft.AspNetCore.Components` be added to the templates' *_Imports.razor* files to make (for example) `ParameterAttribute` available for partial classes?

Either way on that question: As part of the update for namespaces generally that I was going to make, I'll add the namespace to the partial class example in the topic. I'll do that right now so we get it in quickly.

Leave this issue open, as it will close automatically when the PR merges.

**Comment by guardrex** [comment]
Thanks @DataJuggler ... try the last two commits on the PR to see if that would have avoided the confusion ...

https://github.com/aspnet/AspNetCore.Docs/pull/16098

This issue will close automatically when the PR merges.
